adapters/adf/adf.go

```diff
@@ -47,6 +53,28 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapte
 
 		imp.TagID = adfImpExt.MasterTagID.String()
 		validImps = append(validImps, imp)
+
+		// If imps specify priceType they should all be the same. If they differ, only the first one will be used
+		if adfImpExt.PriceType != "" && priceType == "" {
+			priceType = adfImpExt.PriceType
+		}
+	}
+
+	if priceType != "" {
+		requestExt := adfRequestExt{}
+
+		if len(request.Ext) > 0 {
+			if err := json.Unmarshal(request.Ext, &requestExt); err != nil {
+				errors = append(errors, err)
+			}
+		}
+
+		requestExt.PriceType = priceType
+
+		var err error
+		if request.Ext, err = json.Marshal(&requestExt); err != nil {
```

Comment:
At this point, you are remarshalling `requestExt`, which may not be properly populated if the above unmarshal failed. This should be very unlikely, perhaps impossible, but you may want to consider if you want the request to progress through this stage if an error happened.

Reply:
@hhhjort Thanks! It should be be fixed, but I don't like the many nested if's. I would probably have done it a little different in a language I'm more familiar with. If you can give me a hint on how to make it look nicer, please do.

Reply:
Actually this doesn't fix it. If you hit line 69, having encountered and error with unmarshalling the `.ext`, you still drop down and overwrite the original `.ext` on line 75. You would have to break out of the section, perhaps returning an error from `MakeRequests()` at that line, to avoid this issue. I am not terribly concerned with this, as it is something that should really never happen in the first place. And if it does somehow happen, it just means that the `request.ext` you are sending to your server may have missing/corrupt data in it. I really just wanted to highlight it so you could decide if it is a worry for you and change your handling if it is. If you reply that you are good with it as is, or just do nothing, I am still good with approving this PR. Just giving you some time to respond before it is merged.

Reply:
Ack, nevermind, it does fix it. I just did not read the code properly.

Reply:
@hhhjort I was frantically reading trying to understand what I missed :) But I think the code is ugly. Can it be written better? You don't have to reply, I could take that online course on Go if I can just get some additional hours in a day :)

Reply:
I don't have any specific recommendations to make this code prettier, it seems normal enough go to me. Well, one thing that might improve the view...

In the initial loop where you try to find the price type, you are only interested in the first value you find. So when you do find it, you could instead call `setRequestPriceType(request *openrtb2.BidRequest, priceType string)`, which would be the following block of code that sets the price type in `request.ext`.  Then in that function, you could return immediately if the unmarshal fails, making it more obvious that a failure aborts the rest of price type processing. Moving the code to its own function isolates it as its own step, leaving the main function cleaner in its logic. You coud also put the loop to find the price type in the function, making it easier to break out of the loop and further focusing the main function's code.

Reply:
@hhhjort yes, common clean code sense.  I may leave it as is for now if you don't mind. I might get back to this at a later time if I come up with something else that needs to be done in the adapter. Is that OK?

Reply:
That is fine, I have already approved this PR